enhancements/multi-arch/heterogenous-architecture.md

```diff
@@ -0,0 +1,270 @@
+
+### Goals
+
+- Provide a way to install and upgrade clusters with heterogenous architecture node pools. This includes any combination of control plane and worker nodes.
+- Generate manifest listed release payloads for the 4 main architectures (x86, ARM, Power and Z)
+
+### Non-Goals
+
+- Manifest lists support for the internal image-registry, Openshift builds (hence CI will not be supported for heterogenous arch payloads)
+- Heterogenous control plane
+
+## Proposal
+
+The support for installation of a heterogenous architecture cluster and the associated components will be done in phases. The phases range from a bare bones no upgrade support installation to a full fledged installation and upgrade
+capability of a cluster on different topologies. The phases are listed below and details on some components of the phases will be described below. 
+
+Phase 0: Prototype
+- Aimed at delivering a prototype that users can experiment with
+- Aimed at Azure/AWS with the primary supported architectures being x86 + ARM.
+- For supporting ARM nodes on Azure, [hyper-v support][BZ-1949614] is required which is landing in RHEL 8.6 and will be included in RHCOS 4.11
+- oc adm command changes to support manifest list payloads
+- Ability to assemble and publish manifest listed release payloads in a reproducible way
+- Does not include upgrade support
+- Is not intended for production
+
+Phase 1: GA of basic support
+- Release pipeline needs to be setup to construct manifest list based release payloads
+- Upgrade support - CVO/Cincinnati support (heterogenous -> heterogenous payload upgrades only)
+- Does not include support for:
+    - image registry - internal image registry will not support manifest listed images
+    - imagestreams
+
+Phase 2: Full basic functionality
+- Imagestream manifest list support
+- OLM/Operator ecosystem improvements
+- oc debug (and other commands) support for compute architecture
+- support toolbox
+- support for Power and Z architectures
+- Hypershift awareness
+
+Phase 3: Quality of life improvements
+- First class install support - create workers at install time
+- Improve console's content filtering by making it architecture aware
+- Possibly reduce mirroring costs by looking into things like sparse manifest list
+- Upgrade support for homogenous payload -> heterogenous payload and vice versa
+
+### User Stories
+
+As Openshift continues to add to the architecture types that it is able to run on (x86, Power, Z, ARM) it is becoming more obvious that there are clear requirements and use cases for Openshift clusters that contain mixed architecture types.
+This becomes even more of the case once we introduce ARM support as a significant number of use cases detail a x86 master node talking to ARM worker nodes. While ACM could be an option in some cases it does not cover the main ones
+we are seeing that this area would need to address.
```

Comment:
A user story should take a form like:

```
As an enterprise looking to save compute resource cost, i want to run part of my application workload on ARM instead of x86
```

```
As the owner of an application that runs partially on x86 and partially on ARM, I want to be able to run the entire application within a single cluster so I do not need to pay the cost of higher latency between clusters, or worry about the HA-resilience of cross-cluster bridges.
```